src/EnergyPlus/DXCoils.cc

```diff
@@ -9426,7 +9426,7 @@ void CalcDoe2DXCoil(EnergyPlusData &state,
     using Curve::CurveValue;
     auto &HPWHCrankcaseDBTemp = state.dataHVACGlobal->HPWHCrankcaseDBTemp;
     auto &SysTimeElapsed = state.dataHVACGlobal->SysTimeElapsed;
-    auto &TimeStepSys = state.dataHVACGlobal->TimeStepSys;
+    Real64 TimeStepSys = state.dataHVACGlobal->TimeStepSys;
```

Comment:
@amirroth Could you clarify why TimeStepSys is changed and SysTimeElapsed is not? They both are read-only. (My guess is that you used an automated way to make this change throughout and only used TimeStepSys but not SysTimeElapsed, but maybe not)

I had started a branch on which I'm fixing some cppcheck warnings before this PR, and I have made changes to similar lines. After seeing your PR I see our changes differ.

What I had gone with was

```c++
    auto const &SysTimeElapsed = state.dataHVACGlobal->SysTimeElapsed;
    auto const &TimeStepSys = state.dataHVACGlobal->TimeStepSys;
```

One benefit of using this is that it's pretty clear in your IDE what this resolves too.

![image](https://user-images.githubusercontent.com/5479063/222439131-aa7f1741-f976-48c2-b1b2-bd8cef80fe3f.png)

But I'm pretty sure you're going to say we should make a copy of the double instead of taking a const reference, right?